index.js

```diff
@@ -1,29 +1,31 @@
 const Discord = require('discord.js');
-const bot = new Discord.Client();
+const DISCORD_TOKEN = process.env.DISCORD_TOKEN;
 
-bot.on('message', (message) => {
+if (!DISCORD_TOKEN) {
+  throw 'DISCORD_TOKEN environment variable is not set.'
+}
 
-  function DiceRoll(string) {
-    let times = parseInt(string.slice(0,1))
-    let dice = parseInt(string.slice(2))
-    for (let i = 0; i < times; i++) {
-      message.channel.sendMessage(Math.floor(Math.random() * dice) + 1)
-    }
-  }
+const bot = new Discord.Client();
 
-  if (message.content[0] == '!') {
-    message.content = message.content.slice(1)
-    if (message.content == 'handbook') {
+bot.on('message', (message) => {
+  if (message.content[0] === '!') {
+    const messageContent = message.content.slice(1).toLowerCase();
+    const splitContent = messageContent.split(" ");
+    if (messageContent === 'handbook') {
       message.channel.sendMessage('https://drive.google.com/file/d/0B3E_3y6MPuYjQ3cteEtLQm53a1U/view');
-    } else if (message.content == 'help') {
-      message.channel.sendMessage('DnD Bot responds to the following commands: ')
-      message.channel.sendMessage('*!handbook*')
-      message.channel.sendMessage('*!roll (insert dice and rolls here)*')
-    } else if (message.content.split(" ")[0] == 'roll') {
-      let dice = message.content.split(" ")[1]
-      DiceRoll(dice)
+    } else if (messageContent === 'help') {
+      message.channel.sendMessage('DnD Bot responds to the following commands: ');
+      message.channel.sendMessage('*!handbook*');
+      message.channel.sendMessage('*!roll (insert dice and rolls here)*');
+    } else if (splitContent[0] === 'roll') {
+      const [times, dice] = splitContent(" ")[1].split("d").map(d => parseInt(d));
+      const rolls = Array(times).fill(null).map(_ => Math.floor(dice*Math.random() + 1));
```

Comment:
Unnecessarily makes things less readable when the other function did the same shit. 
Its cleaner to have that in a separate function imo.

Reply:
`splitContent` is already an array so  `splitContent(" ")` would fail.

Reply:
Yea, but it wasn't functional enough.  I'm gonna just re-write this shit in scala.